core/iwasm/libraries/lib-socket/inc/wasi_socket_ext.h

```diff
@@ -85,6 +85,20 @@ typedef struct __wasi_addr_info_hints_t {
     uint8_t hints_enabled;
 } __wasi_addr_info_hints_t;
 
+typedef struct __wasi_linger {
```

Comment:
given this isn't going to be a commonly used structure, I wonder if we can just split it into two values rather than defining a new structure? The same about `__wasi_ip_mreq_t`